pkg/operator2/configobservation/configobservercontroller/observe_config_controller.go

```diff
@@ -0,0 +1,89 @@
+package configobservercontroller
+
+import (
+	"k8s.io/client-go/tools/cache"
+
+	configinformers "github.com/openshift/client-go/config/informers/externalversions"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/configobserver/apiserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	"github.com/openshift/cluster-authentication-operator/pkg/operator2/configobservation"
+	"github.com/openshift/cluster-authentication-operator/pkg/operator2/configobservation/console"
+	"github.com/openshift/cluster-authentication-operator/pkg/operator2/configobservation/infrastructure"
+	"github.com/openshift/cluster-authentication-operator/pkg/operator2/configobservation/routersecret"
+)
+
+type ConfigObserver struct {
+	factory.Controller
+}
+
+func NewConfigObserver(
+	operatorClient v1helpers.OperatorClient,
+	kubeInformersForNamespaces v1helpers.KubeInformersForNamespaces,
+	configInformer configinformers.SharedInformerFactory,
+	resourceSyncer resourcesynccontroller.ResourceSyncer,
+	eventRecorder events.Recorder,
+) *ConfigObserver {
+	interestingNamespaces := []string{
+		"openshift-authentication",
+	}
+
+	preRunCacheSynced := []cache.InformerSynced{}
+	for _, ns := range interestingNamespaces {
+		preRunCacheSynced = append(preRunCacheSynced,
+			kubeInformersForNamespaces.InformersFor(ns).Core().V1().ConfigMaps().Informer().HasSynced,
+			kubeInformersForNamespaces.InformersFor(ns).Core().V1().Secrets().Informer().HasSynced,
+		)
+	}
+
+	informers := []factory.Informer{
+		operatorClient.Informer(),
+		configInformer.Config().V1().APIServers().Informer(),
+		configInformer.Config().V1().Consoles().Informer(),
+		configInformer.Config().V1().Infrastructures().Informer(),
+		configInformer.Config().V1().Proxies().Informer(),
+	}
+
+	for _, ns := range interestingNamespaces {
+		informers = append(informers,
+			kubeInformersForNamespaces.InformersFor(ns).Core().V1().ConfigMaps().Informer(),
+			kubeInformersForNamespaces.InformersFor(ns).Core().V1().Secrets().Informer(),
+		)
+	}
+
+	c := &ConfigObserver{
+		Controller: configobserver.NewConfigObserver(
+			operatorClient,
+			eventRecorder,
+			configobservation.Listers{
+				SecretsLister: kubeInformersForNamespaces.SecretLister(),
+
+				APIServerLister_:     configInformer.Config().V1().APIServers().Lister(),
+				ConsoleLister:        configInformer.Config().V1().Consoles().Lister(),
+				InfrastructureLister: configInformer.Config().V1().Infrastructures().Lister(),
+				ResourceSync:         resourceSyncer,
+				PreRunCachesSynced: append(preRunCacheSynced,
+					operatorClient.Informer().HasSynced,
+
```

Comment:
weird, proxy observation is not a part of this PR at all, I wonder where the other proxy informer came from.

Let's do proxy in a follow-up PR so that we can see whether we can get the observation working such that we only need to observe to a single field?